clientupdate/clientupdate.go

```diff
@@ -712,10 +741,36 @@ func (up *Updater) updateWindows() error {
 	panic("unreachable")
 }
 
+func (up *Updater) switchOutputToFile() (io.Closer, error) {
+	var logFilePath string
+	exePath, err := os.Executable()
+	if err != nil {
+		logFilePath = filepath.Join(os.TempDir(), "tailscale-updater.log")
+	} else {
+		logFilePath = strings.TrimSuffix(exePath, ".exe") + ".log"
+	}
+
+	up.Logf("writing update output to %q", logFilePath)
+	logFile, err := os.Create(logFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	up.Logf = func(m string, args ...any) {
+		fmt.Fprintf(logFile, m+"\n", args...)
+	}
+	up.Stdout = logFile
+	up.Stderr = logFile
+	return logFile, nil
+}
+
 func (up *Updater) installMSI(msi string) error {
 	var err error
 	for tries := 0; tries < 2; tries++ {
 		cmd := exec.Command("msiexec.exe", "/i", filepath.Base(msi), "/quiet", "/promptrestart", "/qn")
+		// Don't automatically launch the app after install.
+		// We will launch it explicitly as the current GUI user afterwards.
+		cmd.Env = append(os.Environ(), "TS_NOLAUNCH=true")
```

Comment:
This needs to be on the msiexec command line, as opposed to the environment. Sorry for not clarifying this!

eg you want the end result on the command line to look something like:
`msiexec /i <path-to-msi> /quiet /promptrestart TS_NOLAUNCH=true`